src/prefect/blocks/core.py

```diff
@@ -585,8 +595,34 @@ def _from_block_document(cls, block_document: BlockDocument):
             block_document.block_document_references
         )
 
+        # Due to the way blocks are loaded we can't directly instrument the
+        # `load` method and have the data be about the block document. Instead
+        # this will emit a proxy event for the load method so that block
+        # document data can be included instead of the event being about an
+        # 'anonymous' block.
+        emit_instance_method_called_event(block, "load", successful=True)
+
         return block
 
+    def _event_kind(self) -> str:
+        return "prefect.block"
+
+    def _event_method_called_resources(self, method_name: str) -> ResourceTuple:
+        name = self._block_document_name if self._block_document_name else "anonymous"
+        return (
+            {
+                "prefect.resource.id": f"prefect.block-document.{name}.{method_name}",
+            },
```

Comment:
So in this case, the resource is a block-document _method_ with a related resource of the block type, but I'm thinking that the method name should be in the event name so and then the resource is just the block(-document) itself:

```suggestion
            {
                "prefect.resource.id": f"prefect.block-document.{name}",
            },
```

This way the block can stand is as the remote resource, like a database or a bucket.

Reply:
Aren't automation executions similar to this? The execution is part of the resource id. Curious what the difference in thought is between block method calls and automation action executions.

Reply:
These are the _event_ names (not the resource IDs) for automations:

![image](https://user-images.githubusercontent.com/8194/222282181-455aa4a5-c80d-4271-a2ff-2ed93e8c494e.png)

Reply:
So what I'm suggesting is block methods look like that.  Also, I wonder if we should change the automations to look like:

```
prefect-cloud.automation.action.run-deployment.executed
prefect-cloud.automation.action.send-notification.failed
```

Reply:
But don't the id's have the action number?

Reply:
Those are the resource IDs:

![image](https://user-images.githubusercontent.com/8194/222283323-ed5610ed-37d9-4943-bc0a-4ebc9bdcea8d.png)

Reply:
The `event` should be "what happened", and the `prefect.resource.id` should be "what it happened to"